### ISSUE TITLE
streamline evals

### ISSUE THREAD

**Issue by mivanit** [understanding-search/maze-transformer#200]
Evals are a mess, and these is no unified way to run them.
- [ ] organize existing evals into categories based on type signature
- [ ] single function which takes a model and runs sets of evals
- [ ] fix usage of evals in the codebase to conform to new standard
- [ ] add all evals to training #97 

currently, we have a couple forms of evals:
- `PathEvals`, which take the solution coordinate and generated rollout
  - these are the oldest group
  - there are a couple things in `rollout_evals()` which belong here
- `LOGIT_ATTRIB_TASKS`, which give the model a specific prompt and have it generate a single token
  - these were made for direct logit attribution, and are of the form "can you predict the origin correctly" or "how well do you do on a random non-endpoint in the path"
- "rollout evals", which take the raw tokens produced by a model and compute things about the validity of tokens
  - these are in `rollout_evals()`
- "logit evals" as mentioned in #165, but this might really be part of `LOGIT_ATTRIB_TASKS`
  - would be nice to have a "total probability mass assigned to valid tokens / coordinate tokens" metric, more useful than just raw perplexity

these need to be wrapped into a couple different groups, depending on the inputs. Perhaps:
- `model, mazes` for the task based evals
- `generated_path, correct_path` as in the older `PathEvals`
- `generated_tokens, maze` as in the "rollout evals"
- `sequence_logits, maze` for logit evals #165 

the function `rollout_evals()` is a mess because it exports stuff which was in a jupyter notebook for the UniReps submission, no time to integrate it with the rest of the code right now.
